File: NetCoreGameServer/NetCoreGameServer/Data/Model/Attack.cs
namespace NetCoreGameServer.Data.Model;

public class Attack
{
    public long Timestamp { get; set; }
    public int SourceId { get; set; }
    public int TargetId { get; set; }
    public AttackType Type { get; set; }
    public int Damage { get; set; }
    public bool IsCritical { get; set; }
    public int TargetHealthResult { get; set; }
    public EquippedItemSlot EquippedItemSlot { get; set; }
    public ItemSubType? WeaponType { get; set; }
}

File: next/src/models/Attack.ts
import { AttackType } from "./AttackType";
import Item, { ItemSubType } from "./Item";

export type Attack = {
  timestamp: number;
  sourceId: number;
  targetId: number;
  type: AttackType;
  damage: number;
  isCritical: boolean;
  targetHealthResult: number;
  weaponType: ItemSubType;

  weapon?: Item;
  xSource: number;
  ySource: number;
  xTarget: number;
  yTarget: number;

  isPlayer: boolean;
}